File: JavaScript/7js 2/makeParts.js
/*
|--------------------------------------------------------------------------
| Cut array into smaller parts
|--------------------------------------------------------------------------
|
*/

function makeParts(arr, chunkSize) {
   let res = [];
   for (let i = 0; i < arr.length; i = i + chunkSize) {
      res.push(arr.slice(i, i + chunkSize));
   }
   return res;
}


function makeParts(arr, chunkSize) {
   return [...Array(Math.ceil(arr.length / chunkSize))].map((val) => arr.splice(0, chunkSize));
}


function makeParts(arr, chunkSize) {
   let res = [];
   while (arr.length > 0) {
      res.push(arr.splice(0, chunkSize));
   }
   return res;
}


console.log(makeParts([1, 2, 3, 4, 5], 2), [[1, 2], [3, 4], [5]]);
console.log(makeParts([1, 2, 3], 1), [[1], [2], [3]]);
console.log(makeParts([1, 2, 3, 4, 5], 10), [[1, 2, 3, 4, 5]]);

File: Python/7py 2/make_parts.py
"""
|--------------------------------------------------------------------------
| Cut array into smaller parts
|--------------------------------------------------------------------------
|
"""

def make_parts(arr, chunkSize):
   res = []
   for i in range(0, len(arr), chunkSize):
      res.append(arr[i:i + chunkSize])
   return res


def makeParts(arr, chunkSize):
   return [arr[i:i+chunkSize] for i in range(0, len(arr), chunkSize)]


def make_parts(arr, chunkSize):
   lst = []
   while arr:
      lst.append(arr[0:chunkSize])
      arr = arr[chunkSize:]
   return lst


print(make_parts([1,2,3,4,5], 2), [[1,2],[3,4],[5]])
print(make_parts([1,2,3], 1), [[1],[2],[3]])
print(make_parts([1,2,3,4,5], 10), [[1,2,3,4,5]])